src/internal/m365/collection/exchange/mail_restore.go

```diff
@@ -229,6 +232,34 @@ func setMessageSVEPs(msg models.Messageable) models.Messageable {
 	return msg
 }
 
+func setReplyTos(msg models.Messageable) {
+	var (
+		replyTos          = msg.GetReplyTo()
+		emailAddress      models.EmailAddressable
+		name, address     string
+		sanitizedReplyTos = make([]models.Recipientable, 0)
+	)
+
+	for _, replyTo := range replyTos {
+		emailAddress = replyTo.GetEmailAddress()
+		address = ptr.Val(emailAddress.GetAddress())
+		name = ptr.Val(emailAddress.GetName())
+
+		if isValidEmail(address) || isValidDN(address) {
+			newEmailAddress := models.NewEmailAddress()
+			newEmailAddress.SetAddress(ptr.To(address))
+			newEmailAddress.SetName(ptr.To(name))
+
+			sanitizedReplyTo := models.NewRecipient()
+			sanitizedReplyTo.SetEmailAddress(newEmailAddress)
+
+			sanitizedReplyTos = append(sanitizedReplyTos, sanitizedReplyTo)
+		}
+	}
+
+	msg.SetReplyTo(sanitizedReplyTos)
```

Comment:
Do we need to bailout early if len(replyTos) is 0?

Reply:
@vkamra In that case the for loop will not iterate. Should we even set empty `sanitizedReplyTos` is the question! Let me return and not set empty replyTos